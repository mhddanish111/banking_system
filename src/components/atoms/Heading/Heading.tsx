import React, { ReactNode } from 'react';

type HeadingProps = {
  className?: String;
  children?: ReactNode;
  HeadingType: any;
  ariaLabel?: String;
};

const Heading: React.FC<HeadingProps> = ({
  children,
  className,
  HeadingType,
  ariaLabel,
}) => (
  <HeadingType className={className} aria-label={ariaLabel}>
    {children}
  </HeadingType>
);

export default Heading;
