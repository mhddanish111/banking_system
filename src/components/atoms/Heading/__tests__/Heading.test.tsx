import { shallow } from 'enzyme';
import Heading from '../Heading';

describe('<Heading />', () => {
  const props = {
    className: '',
    ariaLabel: 'heading',
  };
  const wrapper = shallow(
    <Heading HeadingType="h1" {...props}>
      Heading
    </Heading>,
  );
  it('should render the Heading Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
