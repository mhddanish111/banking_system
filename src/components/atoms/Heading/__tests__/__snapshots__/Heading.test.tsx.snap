// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the Heading Component correctly 1`] = `
<h1
  aria-label="heading"
  className=""
>
  Heading
</h1>
`;
